tests/linalg_test.py

```diff
@@ -1620,6 +1620,55 @@ def testToeplitzConstructionWithKnownCases(self):
       [3, 2, 1, 4]], dtype=np.float32))
 
 
+  @jtu.sample_product(
+    shape=[[2, 1], [1, 2], [2, 5], [5, 2], [4, 20], [20, 4]],
+    dtype=float_types + complex_types)
+  def testNullSpace(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    X = rng(shape, dtype)
+    args_maker = lambda: [X]
+    self._CheckAgainstNumpy(jtu.promote_like_jnp(osp.linalg.null_space),
+                            jsp.linalg.null_space, args_maker)
+
+  @jtu.sample_product(
+    shape=[[2, 1], [1, 2], [2, 5], [5, 2], [4, 20], [20, 4]],
+    dtype=float_types + complex_types,
+    size=[None, 2, 9])
+  def testNullSpaceCompiled(self, shape, dtype, size):
+    # test compiled version
+    rng = jtu.rand_default(self.rng())
+    X = rng(shape, dtype)
+    if size != None:
+      def scipy_fun(x):
+        result = osp.linalg.null_space(x)
+        if size <= result.shape[1]:
+          return result[:, -size:]
+        else:
+          all_zeros = np.zeros((result.shape[0], size), dtype=x.dtype)
+          all_zeros[:, size-result.shape[1]:] = result
+          return all_zeros
+      args_maker = lambda: [X]
+      jnp_fun = partial(jsp.linalg.null_space, size=size)
+      self._CheckAgainstNumpy(scipy_fun, jnp_fun, args_maker, check_dtypes=False)
```

Comment:
You could put this `_CheckAgainstNumpy` outside the `if size is not None` block and it would basically cover everything that `testNullSpace` currently covers, right? Can we combine these two tests?

Reply:
agreed!! It obviously should be a single test, now that I understand the different parts it makes sense to group them into one
Fixed that